КАЗАНЦЕВ

SimonKazantsev opened the pull request:
<img width="1919" height="944" alt="изображение" src="https://github.com/user-attachments/assets/836eeba7-f570-4e78-a8e7-9f3ad912537f" />


Случайно сделал не свой вариант, потому делаю новый pull-request

SimonKazantsev committed:
Inital commit

SimonKazantsev committed:
Случайно сделал не свой вариант

SimonKazantsev committed:
исправил диаграмму

dekabrsky commented on .idea/.gitignore:
.idea - в gitignore

dekabrsky commented on src/main/kotlin/Main.kt:
```diff
@@ -1,3 +1,15 @@
+import resolver.Resolver
 fun main(args: Array<String>) {
-    print("Yeah rock!")
+    val parser = CsvParser
```
Раз это object, можно использовать напрямую

dekabrsky commented on src/main/kotlin/model/Player.kt:
```diff
@@ -1,3 +1,18 @@
+    val name: String,
+    val team: String,
+    val city: String,
+    val position: String,
```
Enum - хорошее решение для позиции

dekabrsky commented on src/main/kotlin/parser/CsvParser.kt:
```diff
@@ -1,3 +1,69 @@
\ No newline at end of file
+object CsvParser {
+    var data: Pair<MutableList<Team>, MutableList<Person>> = Pair(mutableListOf(), mutableListOf())
+    private var file: File = File("C:/Users/papak/AndroidStudioProjects/" +
```
Лучше в самой функции

dekabrsky commented on src/main/kotlin/parser/CsvParser.kt:
```diff
@@ -1,3 +1,69 @@
+    fun formation_of_Players_and_Teams(notes: MutableList<List<String>>?):
+            Pair<MutableList<Team>, MutableList<Person>>  {
+        var teams: MutableList<Team> = mutableListOf()
+        var players: MutableList<Person> = mutableListOf()
```
Здесь и везде -  val вместо var

dekabrsky commented on src/main/kotlin/parser/CsvParser.kt:
```diff
@@ -1,3 +1,69 @@
+                participations = it[7].toInt(),
+                goals = it[8].toInt(),
+                assists = it[9].toInt(),
+                yellow_cards = it[10].toInt(),
```
toIntOrNull

dekabrsky commented on src/main/kotlin/parser/CsvParser.kt:
```diff
@@ -1,3 +1,69 @@
+                red_cards = it[11].toInt()
+            )
+            players.add(player)
+            var team = teams.find {it.name == player.team}
```
Можно подумать, как сделать через groupBy

dekabrsky commented on src/main/kotlin/plot/KandyPlot.kt:
```diff
@@ -0,0 +1,34 @@
+
+class KandyPlot {
+    fun main(teams: List<Team>) {
+        val forward_players = teams.flatMap { it.players }
```
посмотри code conventions

dekabrsky commented on src/main/kotlin/resolver/Resolver.kt:
```diff
@@ -1,3 +1,48 @@
+        var average: Double = 0.0
+        var team: Team = teams[0]
+        teams.forEach { it ->
+            var red_cards_by_team: Int = 0
```
groupBy + map + average

SimonKazantsev committed:
действительно добавил в gitignore, CVSparser используется как объект

SimonKazantsev committed:
исправил неточности

# Conflicts:
#	src/main/kotlin/Main.kt
#	src/main/kotlin/model/Player.kt
#	src/main/kotlin/parser/CsvParser.kt
#	src/main/kotlin/plot/KandyPlot.kt
#	src/main/kotlin/resolver/Resolver.kt

SimonKazantsev commented:
Исправил Неточности
